### PR TITLE
added Asset and File options for Image Providers in the package as against having only network image.

### PR DESCRIPTION
**Pull request by samuelezedi**
You have a great package, I tried using it and discovered it only works with network images, so I have implemented asset and file image provider to the package here's an example.

```
ImageStack(
       imageList: [
             'assets/images/user.jpg',
             'assets/images/user.jpg',
              'assets/images/user.jpg',
              'assets/images/user.jpg',
              'assets/images/user.jpg',
            ],
       imageSource: ImageSource.Asset,
       totalCount: 5,
       imageRadius: 30, // Radius of each images
       imageCount: 5, // Maximum number of images to be shown in stack
       imageBorderWidth: 3, // Border width around the images
 )
```

ALSO

Based on this issue https://github.com/karan413255/image_stack/issues/1

I have implement `ImageStack.widgets` and `ImageStack.providers`.

This will enable users place custom widgets and ImageProviders. Look Below to see code samples.

```
ImageStack.widgets(
         children: <Widget>[
              Center(child: Text('1'), ),
              Center(child: Text('1'),),
              Center(child: Text('1'),),
              Center(child: Text('1') ),
              Center(child: Text('1'),),
           ],
           backgroundColor: Colors.black54,
           totalCount: 5,
           widgetRadius: 30, // Radius of each images
           widgetCount: 5, // Maximum number of images to be shown in stack
           widgetBorderWidth: 2,
           widgetBorderColor: Colors.black,// Border width around the images
    )
```

```
ImageStack.providers(
     providers: [
     AssetImage('assets/images/user.jpg'),
     AssetImage('assets/images/user.jpg'),
     FileImage(file.path),
      NetwordImage('http://place.com/assets/images/user.jpg'),
  ],
   backgroundColor: Colors.black54,
     totalCount: 4,
     imageRadius: 30, // Radius of each images
      imageCount: 4, // Maximum number of images to be shown in stack
     imageBorderWidth: 2,
    imageBorderColor: Colors.black,// Border width around the images
 );
```

### CODE REVIEW

**Comment by karan413255** [comment]
Hi @samuelezedi @piperstackist , thanks for contributing. I will check and let you know.

File: lib/image_stack.dart [hunk 1]
```diff
@@ -48,52 +128,110 @@ class ImageStack extends StatelessWidget {
...
+            overflow: Overflow.visible,
+            textDirection: TextDirection.rtl,
+            children: children.length > 0
+                ? widgets
+                : providers.length > 0 ? providersImages : images,
+          )
               : SizedBox(),
           Container(
             margin: EdgeInsets.only(left: 5),
             child: totalCount - images.length > 0
```
Comment (karan413255) [inline lib/image_stack.dart:188]: images.length will always take the images list and not widgets or providersImages to get the remaining count to show.

```suggestion
            child: totalCount - (check for all the list's length to get the remaining count) > 0
```

**Comment by karan413255** [comment]
@samuelezedi please take a look at my review comment

**Comment by samuelezedi** [comment]
@karan413255 Okay looks good.

**Comment by karan413255** [comment]
@samuelezedi can you make the changes for the review comments?

File: lib/image_stack.dart [hunk 2]
```diff
@@ -48,52 +128,110 @@ class ImageStack extends StatelessWidget {
...
-                  overflow: Overflow.visible,
-                  textDirection: TextDirection.rtl,
-                  children: images,
-                )
+            overflow: Overflow.visible,
+            textDirection: TextDirection.rtl,
+            children: children.length > 0
+                ? widgets
+                : providers.length > 0 ? providersImages : images,
+          )
```
Comment (samuelezedi) [inline lib/image_stack.dart:184]: look okay

**Review by samuelezedi — COMMENTED** [review]
Changes looks good

**Review by karan413255 — APPROVED** [review]

**Pull request merged**